src/Controller/GameController.php

```diff
@@ -17,12 +22,107 @@ public function __construct(GameManipulator $gameManipulator)
         $this->gameManipulator = $gameManipulator;
     }
 
-    public function random(): Response
-    {        
-        $game = $this->gameManipulator->createGame();
+    /**
+     * @ParamConverter("game", options={"mapping": {"hash": "hash"}})
+     */
+    public function index(Request $request, Game $game): Response
+    {
+        if ($game->isAbandoned()) {
+            return $this->render('abandoned.html.twig');
+        }
+
+        $abandonForm = $this->createForm(AbandonGameType::class);
+        $abandonForm->handleRequest($request);
+
+        if ($abandonForm->isSubmitted() && $abandonForm->isValid()) {
+            $this->gameManipulator->abandonGame($game);
+
+            return $this->redirectToRoute('index');
+        }
+
+        $triggers = $this->createTriggersForms($game);
+
+        $trigger = $this->createForm(ShootShipType::class)->handleRequest($request);
+
+        if ($trigger->isSubmitted() && $trigger->isValid()) {
+            $data = $trigger->getData();
+
+            $game = $this->getDoctrine()
+                ->getRepository(Game::class)
+                ->findOneByHash($data['game_hash']);
+
+            if (!$game) {
+                throw $this->createNotFoundException(
+                    'No game found for hash '.$data['game_hash']
+                );
+            }
+            
+            $coordinates = explode(',', $data['coordinates']);
+
+            $shoot = $this->gameManipulator->shoot($coordinates, $game);
+
+            $shipHit = $this->gameManipulator->didShootHitOpponentShip($shoot, $game);
+            if ($shipHit) {
+                if ($this->gameManipulator->isShipSunk($shipHit, $game)) {
+                    $this->addFlash('success', 'COULEEEEEEEE!');
```

Comment:
Please translate this messages

Here is a simple example of translation in the controller
https://symfony.com/doc/current/translation.html#basic-translation